jit-service/src/main/java/org/dcsa/jit/service/TimestampRoutingService.java

```diff
@@ -0,0 +1,53 @@
+package org.dcsa.jit.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.dcsa.jit.persistence.entity.MessageRoutingRule;
+import org.dcsa.jit.persistence.entity.OutboxMessage;
+import org.dcsa.jit.persistence.repository.MessageRoutingRuleRepository;
+import org.dcsa.jit.persistence.repository.OutboxMessageRepository;
+import org.dcsa.jit.transferobjects.TimestampTO;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class TimestampRoutingService {
+  private final MessageRoutingRuleRepository messageRoutingRuleRepository;
+  private final OutboxMessageRepository outboxMessageRepository;
+  private final ObjectMapper objectMapper;
+
+  @Transactional
+  @SneakyThrows // JsonProcessingException when serializing timestamp
+  public void routeMessage(TimestampTO timestamp) {
+    String vesselIMONumber;
+    if (timestamp.vessel() != null && timestamp.vessel().vesselIMONumber() != null) {
+      vesselIMONumber = timestamp.vessel().vesselIMONumber(); // jit 1.2
+    } else {
+      vesselIMONumber = timestamp.vesselIMONumber(); // pre jit 1.2
+    }
```

Comment:
The `1.0` field is mandatory, so this if can be simplified to just `timestamp.vesselIMONumber()`

Reply:
I know, it is purposely checking both because that is what we are supposed to do..

Reply:
To be honest, I do not feel I see the point of doing that.  We know the fields must be the same (create will enforce that).  Repeating this here (in a different variant) just makes the code WET for me.

Reply:
Fair point, fixed